buildSrc/private/src/main/kotlin/androidx/build/AndroidXComposeMultiplatformExtensionImpl.kt

```diff
@@ -288,31 +307,37 @@ fun enableOELPublishing(project: Project) {
 
     val ext = project.multiplatformExtension ?: error("expected a multiplatform project")
 
-    ext.targets.all { target ->
-        if (target is KotlinAndroidTarget) {
-            project.publishAndroidxReference(target)
+    val oelGroupId = project.findProperty("oel.androidx.groupId") as? String
+
+    val newRootComponent: CustomRootComponent? = if (oelGroupId != null) {
+        val oelVersion = project.findProperty("oel.androidx.${project.name}.version") as? String
+        requireNotNull(oelVersion) {
+            "Please specify oel.androidx.${project.name}.version property"
         }
+
+        val rootComponent = project
+            .components
+            .withType(KotlinSoftwareComponentWithCoordinatesAndPublication::class.java)
+            .getByName("kotlin")
+
+        val newDependency = project.dependencies.create(oelGroupId, project.name, oelVersion)
+        CustomRootComponent(rootComponent, newDependency)
+    } else {
+        null
     }
-}
 
+    val oelTargetNames = (project.findProperty("oel.publication.targetNames") as? String ?: "")
+        .split(",").toSet()
 
-/**
- * Usage that should be added to rootSoftwareComponent to represent android-specific variants
- * It will be serialized to *.module in "variants" collection.
- */
-private class CustomAndroidUsage(
-    private val name: String,
-    private val attributes: AttributeContainer,
-    private val dependencies: Set<ModuleDependency>
-) : UsageContext {
-    override fun getName(): String = name
-    override fun getArtifacts(): Set<PublishArtifact> = emptySet()
-    override fun getAttributes(): AttributeContainer = attributes
-    override fun getCapabilities(): Set<Capability> = emptySet()
-    override fun getDependencies(): Set<ModuleDependency> = dependencies
-    override fun getDependencyConstraints(): Set<DependencyConstraint> = emptySet()
-    override fun getGlobalExcludes(): Set<ExcludeRule> = emptySet()
-    override fun getUsage(): Usage = error("Should not be accessed!")
+    ext.targets.all { target ->
+        // TODO (o.k): support projects where oel publication is required for both android and native
```

Comment:
We might need it soon, before 1.6-rc01.

It depends whether it affects that we don't declare `android` in collections/annotations any issues in IDE. I will check that in my task https://youtrack.jetbrains.com/issue/COMPOSE-481/skikoMain-API-is-available-in-commonMain

Reply:
changed "desktop" target to "jvm" in the last commit - https://github.com/JetBrains/compose-multiplatform-core/pull/1051/commits/d73e670a7af7064b0804ac7e8b7ca353288feff5

It works fine (published locally and tested on all targets of chat example).